In [4]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

df=pd.read_csv(r"C:\Users\risla\OneDrive\Desktop\EDA PROJECT\DATA\cleaned_day2.csv")
df.head()

Income  Age  Dependents     Occupation City_Tier          Rent  \
0   44637.249636   49           0  Self_Employed    Tier_1  13391.174891   
1   26858.596592   34           2        Retired    Tier_2   5371.719318   
2   50367.605084   35           1        Student    Tier_3   7555.140763   
3  101455.600247   21           0  Self_Employed    Tier_3  15218.340037   
4   24875.283548   52           4   Professional    Tier_2   4975.056710   

   Loan_Repayment    Insurance     Groceries    Transport  ...  \
0        0.000000  2206.490129   6658.768341  2636.970696  ...   
1        0.000000   869.522617   2818.444460  1543.018778  ...   
2     4612.103386  2201.800050   6313.222081  3221.396403  ...   
3     6809.441427  4889.418087  14690.149363  7106.130005  ...   
4     3112.609398   635.907170   3034.329665  1276.155163  ...   

   Potential_Savings_Utilities  Potential_Savings_Healthcare  \
0                   678.292859                     67.682471   
1                   286.668408                      6.603212   
2                   488.383423                      7.290892   
3                  1389.815033                    193.502754   
4                   194.117130                     47.294591   

   Potential_Savings_Education  Potential_Savings_Miscellaneous  \
0                     0.000000                        85.735517   
1                    56.306874                        97.388606   
2                   106.653597                       138.542422   
3                     0.000000                       296.041183   
4                    67.388120                        96.557076   

   Total_Expense  Savings_Rate  Lifestyle_Expense  Essential_Expense  \
0   33371.621929      0.252382        4019.511102       29352.110827   
1   17181.777859      0.360288        2263.854847       14917.923012   
2   36476.154459      0.275801        3865.492522       32610.661938   
3   69837.646632      0.311643       10424.499110       59413.147522   
4   18609.583016      0.251885        1868.524377       16741.058639   

   Extra_Savings  Income_Segment  
0    5065.090514            high  
1    7753.642299          middle  
2    6841.090203            high  
3   14922.988478       very_high  
4    4391.601098          middle  

[5 rows x 33 columns]

## **Hypothesis Testing**

**Do Tier 1 and Tier 3 city people have significantly different savings rates — or is it just by chance?**

**1. T-Test — Savings Rate by City Tier**


In [3]:
from scipy import stats
tier1=df[df['City_Tier']=='Tier_1']['Savings_Rate']
tier2=df[df['City_Tier']=='Tier_2']['Savings_Rate']

t_stat, p_value=stats.ttest_ind(tier1,tier2)
print(f"T-statistic:{t_stat:.4f}")
print(f"P-value:{p_value:.4f}")

if p_value<0.05:
    print('Statistically Significant - City Tier does affect Savings rate')
else:
    print('Due to some random chance - City Tier does not affect Savings rate ')

T-statistic:-76.8626
P-value:0.0000
Statistically Significant - City Tier does affect Savings rate




### 1. T-Test — Savings Rate by City Tier

**Problem Statement:**
Do people living in different city tiers have significantly different savings rates?

**Why this matters:**
Our boxplot showed Tier 3 saves more than Tier 1 — but is this difference 
real or just random chance? T-test confirms this statistically.

**Hypothesis:**
- H0 (Null): There is NO significant difference in savings rate 
  between Tier 1 and Tier 3 city people
- H1 (Alternate): There IS a significant difference in savings rate 
  between Tier 1 and Tier 3 city people

**Significance Level:** α = 0.05

**Result:**
- T-statistic: -76.8626
- P-value: 0.0000

**Conclusion:**
Since p-value (0.0000) < 0.05, we REJECT the null hypothesis.
✅ City Tier significantly affects Savings Rate — Tier 3 people 
save significantly more than Tier 1 people.

**Anova**

**Does Income Segment significantly affect Savings Rate?**

In [6]:
from scipy import stats
low=df[df['Income_Segment']=='low']['Savings_Rate']
middle=df[df['Income_Segment']=='middle']['Savings_Rate']
high=df[df['Income_Segment']=='high']['Savings_Rate']
very_high=df[df['Income_Segment']=='very_high']['Savings_Rate']

f_stat,p_value=stats.f_oneway(low,middle,high,very_high)

print(f"F-statistic:{f_stat:.4f}")
print(f"P-value:{p_value:.4f}")

if p_value<0.05:
    print('statisticaly significant-Income Segment DOES affect Savings Rate')
else:
    print('Due to random chance-Income segement does not affect savings')


F-statistic:1.8738
P-value:0.1316
Due to random chance-Income segement does not affect savings


### 2. ANOVA — Income Segment vs Savings Rate

**Problem Statement:**
Does a person's income level significantly affect their savings rate?

**Hypothesis:**
- H0 (Null): There is NO significant difference in savings rate 
  across income segments
- H1 (Alternate): There IS a significant difference in savings rate 
  across income segments

**Significance Level:** α = 0.05

**Result:**
- F-statistic: 1.8738
- P-value: 0.1316

**Conclusion:**
- Since p-value (0.1316) > 0.05, we FAIL TO REJECT the null hypothesis.
- Income segment does NOT significantly affect savings rate — 
- people save at similar rates regardless of how much they earn.

In [13]:
from scipy.stats import chi2_contingency

contingency_table=pd.crosstab(df['City_Tier'],df['Occupation'])
print(contingency_table)

chi2, p_value, dof, expected= chi2_contingency(contingency_table)

print(f" \nChi_square Statistic:{chi2:.4f}")
print(f"P-value:{p_value:.4f}")
print(f"Degrees of freedom:{dof}")

if p_value<0.05:
    print("statisticaly significant- City Tier and Occupation are related")
else:
    print('Due to some random chance - City Tier and Occupation are not related')

Occupation  Professional  Retired  Self_Employed  Student
City_Tier                                                
Tier_1              1489     1505           1473     1467
Tier_2              2499     2513           2515     2541
Tier_3              1023     1001            979      995
 
Chi_square Statistic:1.6340
P-value:0.9501
Degrees of freedom:6
Due to some random chance - City Tier and Occupation are not related


### 3. Chi-Square — City Tier vs Occupation

**Problem Statement:**
Is there a relationship between which city tier a person lives in 
and their occupation?

**Hypothesis:**
- H0 (Null): City Tier and Occupation are NOT related
- H1 (Alternate): City Tier and Occupation ARE related

**Significance Level:** α = 0.05

**Result:**
- Chi-square Statistic: 1.6340
- P-value: 0.9501
- Degrees of Freedom: 6

**Conclusion:**
- Since p-value (0.9501) > 0.05, we **FAIL TO REJECT** the null hypothesis.
- No significant relationship — City Tier and Occupation are 
- independent of each other.

In [3]:
df.head()

Income  Age  Dependents     Occupation City_Tier          Rent  \
0   44637.249636   49           0  Self_Employed    Tier_1  13391.174891   
1   26858.596592   34           2        Retired    Tier_2   5371.719318   
2   50367.605084   35           1        Student    Tier_3   7555.140763   
3  101455.600247   21           0  Self_Employed    Tier_3  15218.340037   
4   24875.283548   52           4   Professional    Tier_2   4975.056710   

   Loan_Repayment    Insurance     Groceries    Transport  ...  \
0        0.000000  2206.490129   6658.768341  2636.970696  ...   
1        0.000000   869.522617   2818.444460  1543.018778  ...   
2     4612.103386  2201.800050   6313.222081  3221.396403  ...   
3     6809.441427  4889.418087  14690.149363  7106.130005  ...   
4     3112.609398   635.907170   3034.329665  1276.155163  ...   

   Potential_Savings_Entertainment  Potential_Savings_Utilities  \
0                       195.151320                   678.292859   
1                       234.131168                   286.668408   
2                       459.965256                   488.383423   
3                       320.190594                  1389.815033   
4                       187.173750                   194.117130   

   Potential_Savings_Healthcare  Potential_Savings_Education  \
0                     67.682471                     0.000000   
1                      6.603212                    56.306874   
2                      7.290892                   106.653597   
3                    193.502754                     0.000000   
4                     47.294591                    67.388120   

   Potential_Savings_Miscellaneous  Total_Expense  Savings_Rate  \
0                        85.735517   33371.621929      0.252382   
1                        97.388606   17181.777859      0.360288   
2                       138.542422   36476.154459      0.275801   
3                       296.041183   69837.646632      0.311643   
4                        96.557076   18609.583016      0.251885   

   Lifestyle_Expense  Essential_Expense  Lifestyle_Ratio  
0        4019.511102       29352.110827         0.090048  
1        2263.854847       14917.923012         0.084288  
2        3865.492522       32610.661938         0.076746  
3       10424.499110       59413.147522         0.102749  
4        1868.524377       16741.058639         0.075116  

[5 rows x 32 columns]

## Feature Engineering

All 6 engineered features were created in `02_cleaning_preprocessing.ipynb`.

|   | Feature | Description |
|---|---------|-------------|
| 1 | Total_Expense | Sum of all monthly expense columns |
| 2 | Savings_Rate | Disposable Income / Income |
| 3 | Lifestyle_Expense | Eating Out + Entertainment + Miscellaneous |
| 4 | Essential_Expense | Rent + Groceries + Utilities + Healthcare + Transport + Insurance + Loan |
| 5 | Extra_Savings | Disposable Income - Desired Savings |
| 6 | Income_Segment | Income categorized into Low/Middle/High/Very High |

These features were used throughout EDA in notebook 03 and statistical testing in notebook 04.

In [14]:
df.to_csv(r"C:\Users\risla\OneDrive\Desktop\EDA PROJECT\DATA\final_data.csv",index='False')
print('saved')

saved


### **Key Insights from Financial Behaviour Analysis**

#### **1: High Outliers Across Financial Columns**
*Several financial columns contain extreme values. These were not removed because they represent real individuals such as very high earners or financially stressed households, making the dataset realistic rather than noisy.*

#### **2: Income Distribution is Right-Skewed**
*Most individuals fall into moderate income ranges, while a small number of high earners increase the average income (₹41,585) above the median (₹30,185), producing a positively skewed distribution.*

#### **3: Savings Rate Shows Consistency Across People**
*Unlike income, the savings rate forms a near-normal distribution, indicating that most people consistently save roughly 20–35% of what they earn.*

#### **Insight 4: All Occupations Equally Distributed**
*Each occupation category contains nearly the same number of individuals, allowing fair comparison without group imbalance affecting the results.*

#### **5: Higher Income Leads to Higher Spending** 
*Income and expenses move together strongly — when earnings increase, spending also increases at a similar pace, showing lifestyle expansion rather than increased saving.*

#### **6: Age Does Not Influence Financial Behaviour**
*There is almost no correlation between age and income, expenses, or savings. Financial patterns appear lifestyle-based rather than age-based.*

#### **7: Dependents Reduce Ability to Save**
*Individuals with more dependents tend to save less, as household responsibilities reduce the remaining disposable income.*

#### **8: Tier 1 Residents Save the Least**
*People living in Tier 1 cities have the lowest median savings rate (around 19%), likely due to higher cost of living pressures.*

#### **9: Tier 3 Residents Save the Most**
*People in Tier 3 cities achieve the highest savings rate (around 33%), benefiting from lower living expenses and better surplus income.*

#### **10: Location Significantly Impacts Savings**
*Statistical testing confirms that savings differences between city tiers are significant, meaning location genuinely affects financial behaviour.*

#### **11: Income Level Does Not Change Saving Habit**
*Across different income groups, savings rates remain similar, suggesting people adjust their lifestyle as income grows rather than saving a larger percentage.*

#### **12: Occupation and City Tier Are Unrelated**
*No statistical relationship exists between job type and residential city tier, indicating career choice does not determine where individuals live in this dataset.*